Arch-independent demangling and add `gnuv2_demangle` for old g++ projects

AngheloAlf opened the pull request:
I tested it locally and works fine.

I'm unsure about a few things.
- I only added this new demangler to mips and x86, should I add it to any other arch?
- Should `cpp_demangle` be removed from mips? I don't know if there's any mips project that uses a new enough g++ compiler that uses the new mangling abi. Or maybe invert `cwdemangle` and `cpp_demangle` since it is more likely for cw to be used in mips than a new g++.

tomsons26 commented:
I've seen old and new mangling formats in gcc built binaries for Windows and Mac so it's definitely not a old mips binary exclusive thing

AngheloAlf commented:
I refactored the demangling code to be architecture-independent.

Now the user can manually select which demangler they prefer to use. objdiff will still try to guess demangling the symbols by default.

<img width="403" height="347" alt="image" src="https://github.com/user-attachments/assets/929c781f-3772-47a2-8aa5-e93bd672fdb7" />

I also updated the "Tools" -> "Demangle..." pop-up to use the selected demangler instead of leaving it hardcoded to codewarrior.

AngheloAlf commented on objdiff-core/src/arch/ppc/mod.rs:
```diff
@@ -308,18 +308,6 @@ impl Arch for ArchPpc {
-        if name.starts_with('?') {
-            msvc_demangler::demangle(name, msvc_demangler::DemangleFlags::llvm()).ok()
-        } else {
-            name = name.trim_start_matches('.');
```
From what I can see only ppc has this trimming logic.

Do you want to always trim `.` when the demangler is not msvc? Or should it be only be applied to specific demanglers (cpp_demangle, cwdemangle, etc) ?

AngheloAlf replied:
Gotcha.

btw, I'm a bit curious, do you have an example a mangled symbol like this? I would like to take an small look at it.
It is okay if you don't have any, don't worry much about it.

AngheloAlf commented on objdiff-gui/src/views/demangle.rs:
```diff
@@ -12,11 +13,12 @@ pub fn demangle_window(
     appearance: &Appearance,
+    demangler: Demangler,
 ) {
     egui::Window::new("Demangle").open(show).show(ctx, |ui| {
```
As-in, keeping each option independent to each other?

AngheloAlf replied:
Something like this?

<img width="409" height="281" alt="image" src="https://github.com/user-attachments/assets/66f221e7-7f40-47bc-9dae-cacd0ca39c5e" />
<img width="409" height="281" alt="image" src="https://github.com/user-attachments/assets/5060d078-b85f-4c37-a51b-4c38aa93ed67" />
